btx/processing/freqdir.py

```diff
@@ -0,0 +1,740 @@
+import os, csv, argparse
+
+import numpy as np
+from mpi4py import MPI
+
+from matplotlib import pyplot as plt
+from matplotlib import colors
+
+from btx.misc.shortcuts import TaskTimer
+
+from btx.interfaces.ipsana import (
+    PsanaInterface,
+    bin_data,
+    bin_pixel_index_map,
+    retrieve_pixel_index_map,
+    assemble_image_stack_batch,
+)
+
+###########################################
+#John Imports
+from numpy import zeros, sqrt, dot, diag
+from numpy.linalg import svd, LinAlgError
+from scipy.linalg import svd as scipy_svd
+import numpy as np
+
+import time
+
+from datetime import datetime
+currRun = datetime.now().strftime("%y%m%d%H%M%S")
+
+import h5py
+
+from PIL import Image
+
+#writeDirec = "h5writes/"
+#############################################
+
+class FreqDir:
+
+    """Parallel Frequent Directions."""
```

Comment:
It'd be great if we could have a short explanation of what this class is about in the docstring here. Also if we could have more details about the arguments it would help down the line.